core/src/lib.rs

```diff
@@ -53,6 +53,14 @@ impl Position {
             .sqrt()
             .abs()
     }
+
+    fn random() -> Position {
+        let mut rng = rand::thread_rng();
```

Comment:
I wonder why rng had to be mutable. Maybe because it modifies some internal state in the sequence generator?

Reply:
yes, it pulls bytes off the sequence generator, and two things can't do that at the same time